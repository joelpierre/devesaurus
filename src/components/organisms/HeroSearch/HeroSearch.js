import React from 'react';
import styles from './HeroSearch.module.scss';
import Container from '../../core/Container/Container';
import Section from '../../core/Section/Section';
import Row from '../../core/Row/Row';
import Flex from '../../core/Flex/Flex';
import InputField from '../../atoms/InputField/InputField';
import Brand from '../../atoms/Brand/Brand';

const HeroSearch = () => {
  return (
    <Section classes={styles['hero-search']}>
      <Container data-test="component-hero-search">
        <Row>
          <Flex colMd={10} colLg={8} colXl={8} classes="mx-auto">
            <div className={styles['hero-search__wrapper']}>
              <Brand center classes={styles['hero-search__brand']} />
              <InputField
                name="hero-search"
                type="text"
                placeholder="Enter a search term. e.g. HTML"
                classes={styles['hero-search__input']}
              />
            </div>
          </Flex>
        </Row>
      </Container>
    </Section>
  );
};

export default HeroSearch;
