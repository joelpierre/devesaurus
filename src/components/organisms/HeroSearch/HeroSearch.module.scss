@import "~sass/core";

.hero-search {
  $root: &;

  position: relative;
  width: 100%;
  min-height: 500px;
  height: 80vh;
  padding: 0;

  &__wrapper {
    width: 100%;
    height: 100%;
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
    //align-items: center;
    justify-content: center;
    align-content: space-around;
  }

  &__brand {
    padding: 0 0 5rem;
    width: 100%;
    min-width: 300px;
    height: 110px;

    @include tablet-landscape {
      width: 100%;
      max-width: 500px;
    }
  }

  &__input {
    margin-bottom: 10rem !important;

    @include tablet-landscape {
      margin-bottom: 15rem !important;
    }
  }

  &__scrolling-mouse {
    position: absolute;
    top: auto;
    bottom: 2rem;
    left: 50%;
    transform: $centerX;
  }
}
