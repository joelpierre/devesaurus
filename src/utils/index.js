export const updateObject = (oldObject, updatedValues) => ({
  ...oldObject,
  ...updatedValues,
});

export const mapOverACFComponents = components => {
  components.map(component => {
    component.acf_fc_layout = component.acf_fc_layout.replace(
      'WordPressAcf_',
      ''
    );
  });
  return components;
};

export const sortWordObj = word => {
  if (word) {
    delete word.author_id;
    delete word.author;
    delete word.author_nicename;
    delete word.content;
    delete word.date_modified;
    delete word.excerpt;
    delete word.yoast;

    return word;
  }

  return false;
};

export const randTheme = () => {
  const theme = ['brand', 'alpha', 'beta', 'gamma', 'psi', 'omega'];
  const rand = randNumBetween(0, theme.length);
  return theme[rand];
};

export const randNumBetween = (min = 0, max = 10) => {
  return Math.floor(Math.random() * max) + min;
};

export const mapTaxonomyTheme = (slug = 'default') => {
  let theme;

  switch (slug) {
    case 'development':
    case 'web-development':
      theme = 'alpha';
      break;
    case 'graphic-design':
    case 'design':
      theme = 'beta';
      break;
    case 'css':
    case 'css3':
      theme = 'gamma';
      break;
    case 'html':
    case 'html5':
      theme = 'psi';
      break;
    case 'news':
      theme = 'omega';
      break;
    default:
      theme = 'brand';
  }

  return theme;
};

export const breakpoint = {
  is(s) {
    const size = s.trim();
    const sizes = {
      xsmall: '599px',
      small: '600px',
      medium: '900px',
      large: '1200px',
      xlarge: '1800px',
    };

    // console.log(window.matchMedia);

    // Eslint doesn't like you accessing hasOwnProperty directly on object.
    // https://eslint.org/docs/rules/no-prototype-builtins
    if (Object.prototype.hasOwnProperty.call(sizes, size)) {
      return window.matchMedia(`only screen and (min-width: ${sizes[size]})`)
        .matches;
    }

    throw new ReferenceError(
      `The size ${size} is not a valid breakpoint: ${JSON.stringify(sizes)}`
    );
  },
};
