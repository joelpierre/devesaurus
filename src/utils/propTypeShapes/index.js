import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  word: PropTypes.string.isRequired,
  origin: PropTypes.string.isRequired,
  syllables: PropTypes.shape({
    count: PropTypes.number.isRequired,
    list: PropTypes.instanceOf(Array).isRequired,
  }).isRequired,
  pronunciation: PropTypes.string,
  results: PropTypes.arrayOf(PropTypes.shape({
    definition: PropTypes.string.isRequired,
    partOfSpeech: PropTypes.oneOf(['noun', 'verb', 'adjective', 'pronoun', 'adverb', 'preposition', 'conjunction', 'interjection']),
  })).isRequired,
  category: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  }),
  tags: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
  })),
};
