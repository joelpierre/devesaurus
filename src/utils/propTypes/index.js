import PropTypes from 'prop-types';

export const wordPropTypeShape = {
  title: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  terms: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      taxonomy: PropTypes.string.isRequired,
    })
  ),
  acf: PropTypes.shape({
    definition: PropTypes.string,
    origin: PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    }),
    syllables: PropTypes.shape({
      count: PropTypes.string.isRequired,
      list: PropTypes.arrayOf(
        PropTypes.shape({
          item: PropTypes.string,
        })
      ).isRequired,
    }).isRequired,
    pronunciation: PropTypes.string,
  }),
};

export const columnShape = PropTypes.oneOf([
  '1',
  1,
  '2',
  2,
  '3',
  3,
  '4',
  4,
  '5',
  5,
  '6',
  6,
  '7',
  7,
  '8',
  8,
  '9',
  9,
  '10',
  10,
  '11',
  11,
  '12',
  12,
]);

export const defaultSize = (size = 'md') => ({
  size,
});

export const sizePropType = {
  size: PropTypes.oneOf(['xs', 'sm', 'md', 'lg', 'xl', 'full']),
};

export const defaultTheme = (theme = 'brand') => ({
  theme,
});

export const themePropType = {
  theme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
    'gradient-brand',
  ]),
};

export const defaultPageTheme = (theme = 'brand') => ({
  pageTheme: theme,
});

export const pageThemePropType = {
  pageTheme: PropTypes.oneOf([
    'brand',
    'alpha',
    'beta',
    'gamma',
    'psi',
    'omega',
    'tint-alpha',
    'tint-beta',
    'tint-gamma',
    'tint-psi',
    'tint-omega',
  ]),
};
