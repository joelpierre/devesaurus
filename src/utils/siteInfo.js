import React from 'react';
import { graphql, StaticQuery } from 'gatsby';

const SiteInfo = () => (
  <StaticQuery
    query={graphql`
      {
        allWordpressSiteMetadata {
          edges {
            node {
              name
              description
            }
          }
        }
      }
    `}
    render={({ allWordpressSiteMetadata }) => (
      <>
        {allWordpressSiteMetadata.edges[0].node.name}
        <span>{allWordpressSiteMetadata.edges[0].node.description}</span>
      </>
    )}
  />
);

export default SiteInfo;
