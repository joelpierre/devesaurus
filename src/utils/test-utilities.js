import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';

import rootReducer from '../store/reducers';
// import { sagaMiddleware } from '../store/configureStore';

// /**
//  * Store Factory
//  * @param initialState
//  * @returns {Store<any, Action> & *}
//  */
// export const storeFactory = (initialState) => {
//   const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
//   return createStoreWithMiddleware(rootReducer, initialState);
// };


/**
 * Return node(s) with the given data-test attribute.
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper.
 * @param {string} val - Value of the data-test attribute.
 * @returns {ShallowWrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};


/**
 * Check the props for a component
 * @param component
 * @param conformingProps
 */
export const checkProps = (component, conformingProps) => {
  const propError = checkPropTypes(
    component.propTypes,
    conformingProps,
    'prop',
    component.name,
  );
  expect(propError)
    .toBeUndefined();
};
