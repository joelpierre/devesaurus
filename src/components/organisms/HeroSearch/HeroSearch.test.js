import React from 'react';
import { shallow } from 'enzyme';

import { findByTestAttr } from '../../../utils/test-utilities';

import HeroSearch from './HeroSearch';

const defaultProps = {};

/**
 * Factory function to create a ShallowWrapper for the HeroSearch component.
 * @param {object} props - Component props specific to setup
 * @returns {ShallowWrapper}
 */
const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<HeroSearch {...setupProps}/>);
};

describe('<HeroSearch/>', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = setup();
  });

  it('Renders the HeroSearch Component without errors', () => {
    const component = findByTestAttr(wrapper, 'component-hero-search');
    expect(component.length)
      .toBe(1);
  });


});
