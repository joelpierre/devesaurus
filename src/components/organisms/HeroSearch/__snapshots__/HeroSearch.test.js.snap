// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroSearch/> should match snapshot 1`] = `
<Section
  classes="hero-search"
  contrast={false}
  theme={null}
  type="section"
>
  <Container
    classes={null}
    data-test="component-hero-search"
    fluid={false}
  >
    <Row
      classes={null}
      column={false}
      reverse={false}
      row={true}
    >
      <Flex
        auto={false}
        classes="mx-auto"
        colLg={8}
        colMd={10}
        colXl={8}
        lg={false}
        md={false}
        sm={false}
        type="div"
        xl={false}
        xs={false}
      >
        <div
          className="hero-search__wrapper"
        >
          <Brand
            center={true}
            classes="hero-search__brand"
            left={true}
            right={false}
            type="logo"
          />
          <InputField
            classes="hero-search__input"
            disabled={false}
            label={null}
            lg={false}
            max={null}
            md={false}
            min={null}
            name="hero-search"
            placeholder="Enter a search term. e.g. HTML"
            sm={false}
            type="text"
          />
        </div>
      </Flex>
    </Row>
  </Container>
</Section>
`;
